src/access_nri_intake/source/builders.py

```diff
@@ -425,7 +425,10 @@ class AccessOm3Builder(BaseBuilder):
     """Intake-ESM datastore builder for ACCESS-OM3 COSIMA datasets"""
 
     PATTERNS = [
-        rf"[^\.]*\.{PATTERNS_HELPERS['om3_components']}\..*({PATTERNS_HELPERS['ymds']}|{PATTERNS_HELPERS['ymd']}|{PATTERNS_HELPERS['ym']})$",  # ACCESS-OM3
+        rf"[^\.]*\.{PATTERNS_HELPERS['om3_components']}\..*({PATTERNS_HELPERS['ymds']}|{PATTERNS_HELPERS['ymd']}|{PATTERNS_HELPERS['ym']})(?:$|{PATTERNS_HELPERS['not_multi_digit']}.*)",  # ACCESS-OM3
+        rf"[^\.]*\.{PATTERNS_HELPERS['om3_components']}\..*({PATTERNS_HELPERS['y']})(?:$|{PATTERNS_HELPERS['not_multi_digit']}.*)",  # ACCESS-OM3
+        # rf"[^\.]*\.{PATTERNS_HELPERS['om3_components']}\..*({PATTERNS_HELPERS['ymds']}|{PATTERNS_HELPERS['ymd']}|{PATTERNS_HELPERS['ym']})(?:$|{PATTERNS_HELPERS['not_multi_digit']}.*)",  # ACCESS-OM3
+        # rf"[^\.]*\.{om3_components}\..*[\.,_]({ymds}|{ymd}|{ym}|{y})(?:$|-{not_multi_digit}.*)" {PATTERNS_HELPERS['y']}
```

Comment:
```suggestion
        rf"[^\.]*\.{PATTERNS_HELPERS['om3_components']}\..*?({PATTERNS_HELPERS['ymds']}|{PATTERNS_HELPERS['ymd']}|{PATTERNS_HELPERS['ym']}|{PATTERNS_HELPERS['y']})(?:$|{PATTERNS_HELPERS['not_multi_digit']}.*)",  # ACCESS-OM3
```

@marc-white It seems like a non-greedy match can resolve this issue. I've quickly tested some scenarios,

```python
import re
from access_nri_intake.source.builders import AccessOm3Builder

b = re.search(AccessOm3Builder.PATTERNS[0], "access-om3.mom6.3d.uh.1mon.mean.1900.nc")
print(b)
print(b.group(0))
print(b.group(1))

b = re.search(AccessOm3Builder.PATTERNS[0], "access-om3.mom6.3d.uh.1mon.mean.1900-01-01-00000.nc")
print(b)
print(b.group(0))
print(b.group(1))

b = re.search(AccessOm3Builder.PATTERNS[0], "access-om3.mom6.3d.uh.1mon.mean.1900-01.nc")
print(b)
print(b.group(0))
print(b.group(1))

b = re.search(AccessOm3Builder.PATTERNS[0], "GMOM_JRA_WD.ww3.hi.1900-01-03-00000.nc")
print(b)
print(b.group(0))
print(b.group(1))

b = re.search(AccessOm3Builder.PATTERNS[0], "GMOM_JRA_WD.ww3.hi.1900.nc")
print(b)
print(b.group(0))
print(b.group(1))

b = re.search(AccessOm3Builder.PATTERNS[0], "GMOM_JRA_WD.ww3.hi.1900-01.nc")
print(b)
print(b.group(0))
print(b.group(1))
```
gives me outputs,
```
$ python test.py
<re.Match object; span=(0, 39), match='access-om3.mom6.3d.uh.1mon.mean.1900.nc'>
access-om3.mom6.3d.uh.1mon.mean.1900.nc
1900
<re.Match object; span=(0, 51), match='access-om3.mom6.3d.uh.1mon.mean.1900-01-01-00000.>
access-om3.mom6.3d.uh.1mon.mean.1900-01-01-00000.nc
1900-01-01-00000
<re.Match object; span=(0, 42), match='access-om3.mom6.3d.uh.1mon.mean.1900-01.nc'>
access-om3.mom6.3d.uh.1mon.mean.1900-01.nc
1900-01
<re.Match object; span=(0, 38), match='GMOM_JRA_WD.ww3.hi.1900-01-03-00000.nc'>
GMOM_JRA_WD.ww3.hi.1900-01-03-00000.nc
1900-01-03-00000
<re.Match object; span=(0, 26), match='GMOM_JRA_WD.ww3.hi.1900.nc'>
GMOM_JRA_WD.ww3.hi.1900.nc
1900
<re.Match object; span=(0, 29), match='GMOM_JRA_WD.ww3.hi.1900-01.nc'>
GMOM_JRA_WD.ww3.hi.1900-01.nc
1900-01
```

Reply:
@minghangli-uni we'll need to add your test filenames into the proper test suite before we can merge.

Reply:
@minghangli-uni I've implemented your suggestion and added some file name test cases - three of those filenames aren't giving the expected test output for the file frequency (I think). I'll need to come back next week with a fresh mind to confirm I haven't just set up the test wrong.